### PR TITLE
MAINT: Handle XML error when reading XmpInformation

### LINKED ISSUE #585: Unexpected xml.parsers.expat.ExpatError on malformed PDF
When running the following code with the latest pypi version of PyPDF2 on the attached input results in an unexpected `xml.parsers.expat.ExpatError`:

## MCVE: Code + PDF

Example document: [test.pdf](https://github.com/mstamy2/PyPDF2/files/5536984/test.pdf)

```python
from PyPDF2 import PdfReader

reader = PdfReader("test.pdf")
reader.xmp_metadata
```

## Traceback

```python
Traceback (most recent call last):
  File "foo.py", line 5, in <module>
    reader.xmp_metadata
  File "/home/moose/Github/py-pdf/PyPDF2/PyPDF2/_reader.py", line 318, in xmp_metadata
    return self.trailer[TK.ROOT].xmp_metadata  # type: ignore
  File "/home/moose/Github/py-pdf/PyPDF2/PyPDF2/generic.py", line 671, in xmp_metadata
    metadata = XmpInformation(metadata)
  File "/home/moose/Github/py-pdf/PyPDF2/PyPDF2/xmp.py", line 206, in __init__
    doc_root: Document = parseString(self.stream.get_data())
  File "/home/moose/.pyenv/versions/3.6.15/lib/python3.6/xml/dom/minidom.py", line 1968, in parseString
    return expatbuilder.parseString(string)
  File "/home/moose/.pyenv/versions/3.6.15/lib/python3.6/xml/dom/expatbuilder.py", line 925, in parseString
    return builder.parseString(string)
  File "/home/moose/.pyenv/versions/3.6.15/lib/python3.6/xml/dom/expatbuilder.py", line 223, in parseString
    parser.Parse(string, True)
xml.parsers.expat.ExpatError: not well-formed (invalid token): line 53, column 15
```

## Environment

```
$ python -c "import PyPDF2; print(PyPDF2.__version__)"
2.3.1-dev
```

### PR DESCRIPTION
**Pull request by MartinThoma**
Closes #585

### CODE REVIEW

**Comment by phoewass** [comment]
It would be nice if `reader.xmp_metadata` can have a non-strict mode.

**Comment by MartinThoma** [comment]
I guess "None" would then be a reasonable return value?

**Comment by phoewass** [comment]
Absolutely!

And thank you for maintaining PyPDF2, it looks in a way better shape.

File: PyPDF2/xmp.py [hunk 1]
```diff
@@ -199,11 +202,17 @@ class XmpInformation(PdfObject):
...
     """
 
     def __init__(self, stream: ContentStream) -> None:
         self.stream = stream
-        doc_root: Document = parseString(self.stream.get_data())
+        try:
+            data = self.stream.get_data()
+            doc_root: Document = parseString(data)
+        except ExpatError:
+            raise PdfReadError("XML in XmpInformation was invalid")
```
Comment (guillaume-uH57J9) [inline PyPDF2/xmp.py:215]: Consider including diagnostic information contained in the ExpactError object (line, offset, kind of XML issue)/

Example:

```
except ExpatError as e:
    print("Got some XML error. " + str(e))
```

**Comment by MartinThoma** [comment]
Adding a non-strict mode (that passes the value down from the reader) touches more things than I initially thought. I think for the moment I'll leave this out.

**Pull request merged**